archlinuxcn/naiveproxy/PKGBUILD

```diff
@@ -118,48 +113,43 @@ build() {
     use_nss_certs=false
   '
 
-  # use system clang
-  # disable clang plugins
-  # build without afdo.prof
-  # flags="$flags"'
-  #   clang_base_path=""
-  #   clang_use_chrome_plugins=false
-  #   clang_use_default_sample_profile=false'
+  # Disable CFI icall for linux x64
+  # See https://github.com/llvm/llvm-project/issues/86430
+  flags="$flags"'
+    use_cfi_icall=false'
 
   rm -rf "./$out"
   mkdir -p out
 
   export DEPOT_TOOLS_WIN_TOOLCHAIN=0
 
-  gn gen "$out" --args="$flags $EXTRA_FLAGS" --script-executable=$PYTHON
+  gn gen "$out" "--args=$flags $EXTRA_FLAGS" "--script-executable=$PYTHON"
 
   ninja -C "$out" naive
-
-  popd
 }
 
-#check() {
-#  SRC_DIR="${srcdir}/${pkgname}-${_pkgver}-${_pkgrel}"
-#  script_dir="${SRC_DIR}/tests"
-#  naive="${SRC_DIR}/src/out/Release/naive"
-#
-#  cd /tmp
-#  python "${script_dir}/basic.py" --naive="$naive"
-#}
+check() {
+  SRC_DIR="${srcdir}/${pkgname}-${_pkgver}-${_pkgrel}"
+  script_dir="${SRC_DIR}/tests"
+  naive="${SRC_DIR}/src/out/Release/naive"
+
+  cd /tmp
+  python "${script_dir}/basic.py" --naive="$naive"
+}
 
 package() {
   pushd "${srcdir}"
-  install -Dm644 naiveproxy.service ${pkgdir}/usr/lib/systemd/system/naiveproxy.service
-  install -Dm644 naiveproxy@.service ${pkgdir}/usr/lib/systemd/system/naiveproxy@.service
-  install -Dm644 naiveproxy.sysusers ${pkgdir}/usr/lib/sysusers.d/naiveproxy.conf
+  install -Dm644 naiveproxy.service "${pkgdir}/usr/lib/systemd/system/naiveproxy.service"
+  install -Dm644 naiveproxy@.service "${pkgdir}/usr/lib/systemd/system/naiveproxy@.service"
+  install -Dm644 naiveproxy.sysusers "${pkgdir}/usr/lib/sysusers.d/naiveproxy.conf"
   popd
 
   pushd "${srcdir}/${pkgname}-${_pkgver}-${_pkgrel}"
-  install -d -m750 -o 0 -g 287 ${pkgdir}/etc/naiveproxy
-  install -Dm644 src/config.json ${pkgdir}/etc/naiveproxy/config.json
-  install -Dm755 src/out/Release/naive ${pkgdir}/usr/bin/naiveproxy
-  install -Dm644 README.md ${pkgdir}/usr/share/doc/naiveproxy/README.md
-  install -Dm644 USAGE.txt ${pkgdir}/usr/share/doc/naiveproxy/USAGE.txt
-  install -Dm644 LICENSE ${pkgdir}/usr/share/licenses/naiveproxy/LICENSE
+  install -d -m750 -o 0 -g 287 "${pkgdir}/etc/naiveproxy"
+  install -Dm644 src/config.json "${pkgdir}/etc/naiveproxy/config.json"
```

Comment:
```log
naiveproxy E: File (etc/naiveproxy) is owned by root:287
naiveproxy W: File (etc/naiveproxy) does not have the world readable bit set.
naiveproxy W: Directory (etc/naiveproxy) does not have the world executable bit set.
```

Reply:
This is intentional.
This is to ensure that the proxy configuration file cannot be accessed by arbitrary programs.